components/my-account/User/UserForm/UserForm.tsx

```diff
@@ -0,0 +1,368 @@
+import React, { useEffect, useState } from 'react'
+
+import { yupResolver } from '@hookform/resolvers/yup'
+import CheckIcon from '@mui/icons-material/Check'
+import ClearIcon from '@mui/icons-material/Clear'
+import { Button, CircularProgress, Grid, useMediaQuery, useTheme } from '@mui/material'
+import getConfig from 'next/config'
+import { useTranslation } from 'next-i18next'
+import { useForm, Controller } from 'react-hook-form'
+import * as yup from 'yup'
+
+import { KiboRadio, KiboSwitch, KiboTextBox } from '@/components/common'
+import { useAuthContext } from '@/context'
+import {
+  useAddRoleToCustomerB2bAccountMutation,
+  useCreateCustomerB2bUserMutation,
+  useDeleteB2bAccountRoleMutation,
+  useUpdateCustomerB2bUserMutation,
+} from '@/hooks'
+import { buildB2bUserRoleParams } from '@/lib/helpers/buildB2bUserRoleParams'
+import { buildCreateCustomerB2bUserParams } from '@/lib/helpers/buildCreateCustomerB2bUserParams'
+import { buildUpdateCustomerB2bUserParams } from '@/lib/helpers/buildUpdateCustomerB2bUserParams'
+
+import { B2BUser } from '@/lib/gql/types'
+
+interface UserFormProps {
+  isEditMode: boolean
+  b2BUser?: B2BUser | any
+  closeUserForm: () => void
+}
+
+export const useFormSchema = () => {
+  const { t } = useTranslation('common')
+  return yup.object({
+    emailAddress: yup
+      .string()
+      .required(t('no-email-error'))
+      .matches(/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i, t('invalid-email-error')),
+    firstName: yup.string().required(t('firstname-error')),
+    lastName: yup.string().required(t('lastname-error')),
+  })
+}
+
+const UserForm = (props: UserFormProps) => {
+  const { isEditMode, b2BUser, closeUserForm } = props
+
+  const { publicRuntimeConfig } = getConfig()
+
+  const { t } = useTranslation('common')
+  const theme = useTheme()
+  const mdScreen = useMediaQuery(theme.breakpoints.up('md'))
+  const userSchema = useFormSchema()
+  const { user } = useAuthContext()
+  const userRoles = publicRuntimeConfig.b2bUserRoles
+  const userFormRadioOptions = publicRuntimeConfig.userFormRadioOptions
+  const { createCustomerB2bUser } = useCreateCustomerB2bUserMutation()
+  const { addRoleToCustomerB2bAccount } = useAddRoleToCustomerB2bAccountMutation()
+  const { updateCustomerB2bUser } = useUpdateCustomerB2bUserMutation()
+  const { deleteB2bAccountUserRole } = useDeleteB2bAccountRoleMutation()
+
+  const [isLoading, setLoading] = useState(false)
+  const {
+    getValues,
+    setValue,
+    handleSubmit,
+    formState: { errors },
+    control,
+    reset,
+  } = useForm({
+    defaultValues: { role: '', emailAddress: '', firstName: '', lastName: '', isActive: true },
+    resolver: yupResolver(userSchema),
+  })
+  const onSubmit = async () => {
+    if (isLoading) return
+    setLoading(true)
+    const formValues = getValues()
+    if (isEditMode) {
+      const variables = buildUpdateCustomerB2bUserParams({ user, b2BUser, values: formValues })
```

Comment:
we can use a single file for this helper function for create and update right ?

Reply:
@kibo-chandradeeptalaha I have created separate files because for each helper there was a single file.
For ex: Create/Delete Wishlist Item